src/plugins/validation/2and3/semantic-validators/items-required-for-array-objects.js

```diff
@@ -10,8 +10,24 @@
 
 const { walk } = require('../../../utils');
 const MessageCarrier = require('../../../utils/messageCarrier');
+const at = require('lodash/at');
 
-module.exports.validate = function({ jsSpec }) {
+const reduceObj = function(jsSpec, obj) {
+  if (obj['$ref']) {
+    var objPath = obj['$ref'].split('/');
+    objPath.shift();
+    return reduceObj(jsSpec, at(jsSpec, [objPath])[0]);
+  } else if (obj['allOf']) {
```

Comment:
What if the object is a `oneOf` or `anyOf`? I'm not sure what you would want to do here but I'm not sure it would work as intended if that happened.

Reply:
I think it might also be good to have a test for the scenario of an `anyOf` here to make sure its covered.

Thinking about it now, I wonder if using the `resolvedSpec` would make all of this go away haha

Reply:
I took a look at using resolvedSpec, but it seems to resolve just $refs and not allOf/oneOf/anyOf unless i'm using it wrong. In this case I dont think oneOfs or anyOfs can be feasibly "reduced" here since for allOf i can just combine all the properties of the children but for oneOfs and anyOfs, each child must be checked that it defines the required property.

Reply:
That's correct, it would resolve all $refs, including the refs inside of allOf/oneOf/anyOf, but no it wouldn't reconcile all of the sub-schemas. It still might be helpful.

I get that's the trouble with oneOf/anyOf, but I'm not sure the solution should be to ignore them. Maybe this function should be modified to return an array of schemas and in the case of single schemas or allOf, you just return one element. Then `forEach` can be called on the result. Just an idea. I guess this would only happen if sub-schemas to an allOf/oneOf/anyOf were _themselves_ a allOf/oneOf/anyOf, at which point the schema is just obnoxiously complicated... but I can't help but feel we should be complete here, even for the extreme scenario